azimuth/modules/base_classes/module.py

```diff
@@ -31,11 +32,11 @@ def __init__(
     ):
         mod_options = mod_options or ModuleOptions()
         self.mod_options = mod_options
-        if diff := (
-            set(self.mod_options.no_alias_dict(exclude_defaults=True).keys())
-            - self.allowed_mod_options
-        ):
+        defined_mod_options = set(self.mod_options.no_alias_dict(exclude_defaults=True).keys())
+        if diff := (defined_mod_options - self.allowed_mod_options - self.required_mod_options):
             raise ValueError(f"Unexpected mod_options {diff} for {self.__class__.__name__}.")
+        if self.required_mod_options and self.required_mod_options.isdisjoint(defined_mod_options):
```

Comment:
I think we should check if ALL options in `required_mod_options` are in `required_mod_options`. We can use `issubset` or `<=`.
```suggestion
        if not self.required_mod_options.issubset(defined_mod_options):
```
The `isdisjoint` will pass as soon as ONE option in `required_mod_options` is in `required_mod_options`. For example, here, `2` is missing.
```python
>>> {1, 2}.isdisjoint({1})
False
```

Reply:
Oh, right! Thank you!